### PR TITLE
Introduce TransactionConfig parameter to query config

### PR DESCRIPTION
**Pull request by thelonelyvulpes**
Exposes Transaction config for driver queries.
- Makes TransactionConfig constructor public.
- transactionConfig added to QueryConfig

### CODE REVIEW

File: Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs [hunk 1]
```diff
@@ -16,63 +16,61 @@
...
+    /// zero.
     /// </exception>
     public TimeSpan? Timeout
     {
         get => _timeout;
-        internal set
+        init
         {
-            if ((value ?? TimeSpan.Zero) < TimeSpan.Zero)
+            if (!value.HasValue || value.Value == TimeSpan.MaxValue)
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs:52]: Is this correct? If value doesn't have a value you are still using it to set _timeout...?

File: Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs [hunk 2]
```diff
@@ -115,42 +113,47 @@ public sealed class TransactionConfigBuilder
...
     public TransactionConfigBuilder WithTimeout(TimeSpan? timeout)
     {
-        _config.Timeout = FixSubmilliseconds(timeout);
-        return this;
-    }
-
-    private TimeSpan? FixSubmilliseconds(TimeSpan? timeout)
-    {
-        if(timeout == null || timeout == TimeSpan.MaxValue)
+        if (!timeout.HasValue || timeout.Value == TimeSpan.MaxValue)
```
Comment (AndyHeap-NeoTech) [inline Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs:132]: Same question as above.

**Review by AndyHeap-NeoTech — APPROVED** [review]
Just a couple of questions. Other than that looks good.

**Review by AndyHeap-NeoTech — APPROVED** [review]
Just a couple of questions. Other than that looks good.

File: Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs (hunk 1 above)
Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs:52]: that is the current behaviour, it is setting null to null so slightly redundant but it meant slightly fewer lines of code.

File: Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs (hunk 2 above)
Comment (thelonelyvulpes) [reply Neo4j.Driver/Neo4j.Driver/TransactionConfig.cs:132]: Same answer as above.

**Pull request merged**